app/livechat/server/hooks/sendToFacebook.ts

```diff
@@ -1,11 +1,32 @@
 import { callbacks } from '../../../../lib/callbacks';
-import { settings } from '../../../settings';
+import { settings } from '../../../settings/server/index';
 import OmniChannel from '../lib/OmniChannel';
 import { normalizeMessageFileUpload } from '../../../utils/server/functions/normalizeMessageFileUpload';
 
+type MessageData = {
+	token: string;
+	file: string;
+	editedAt: Date;
+	msg: string;
+	t: string;
+};
+
+type RoomData = {
+	editedAt: Date;
+	t: string;
+	facebook: {
+		page: {
+			id: string;
+		};
+	};
+	v: {
+		token: string;
+	};
+};
+
 callbacks.add(
 	'afterSaveMessage',
-	function (message, room) {
+	function (message: MessageData, room: RoomData | undefined) {
```

Comment:
Refer to prev comments on other hook files

Reply:
Just created the missing properties in IOmnichannelRoom and the bug disappeared:

```
facebook?: {
		page: {
			id: string;
		};
	};
```

Implemented in 7a13094e9650da828ed0162cea644c4e822c58b3